examples/fluids/src/misc.c

```diff
@@ -465,3 +466,136 @@ PetscErrorCode IntArrayP2C(PetscInt num_entries, PetscInt **array_petsc, CeedInt
+  // PETSc
+  char box_faces_str[PETSC_MAX_PATH_LEN] = "3,3,3";
+  if (problem->dim == 2) box_faces_str[3] = '\0';
+  PetscCall(PetscOptionsGetString(NULL, NULL, "-dm_plex_box_faces", box_faces_str, sizeof(box_faces_str), NULL));
+  MatType mat_type;
+  VecType vec_type;
+  PetscCall(DMGetMatType(user->dm, &mat_type));
+  PetscCall(DMGetVecType(user->dm, &vec_type));
+  PetscCall(PetscPrintf(comm,
+                        "  PETSc:\n"
+                        "    Box Faces                          : %s\n"
+                        "    DM MatType                         : %s\n"
+                        "    DM VecType                         : %s\n"
+                        "    Time Stepping Scheme               : %s\n",
+                        box_faces_str, mat_type, vec_type, phys_ctx->implicit ? "implicit" : "explicit"));
+  if (user->app_ctx->cont_steps) {
+    PetscCall(PetscPrintf(comm,
+                          "  Continue:\n"
+                          "    Filename:                          : %s\n"
+                          "    Step:                              : %" PetscInt_FMT "\n"
+                          "    Time:                              : %g\n",
+                          user->app_ctx->cont_file, user->app_ctx->cont_steps, user->app_ctx->cont_time));
+  }
+  // Mesh
+  const PetscInt num_comp_q = 5;
+  PetscInt       glob_dofs, owned_dofs, local_dofs;
+  const CeedInt  num_P = user->app_ctx->degree + 1, num_Q = num_P + user->app_ctx->q_extra;
+  // -- Get global size
+  PetscCall(DMGetGlobalVectorInfo(user->dm, &owned_dofs, &glob_dofs, NULL));
+  // -- Get local size
+  PetscCall(DMGetLocalVectorInfo(user->dm, &local_dofs, NULL, NULL));
+  PetscCall(PetscPrintf(comm,
+                        "  Mesh:\n"
+                        "    Number of 1D Basis Nodes (P)       : %" CeedInt_FMT "\n"
+                        "    Number of 1D Quadrature Points (Q) : %" CeedInt_FMT "\n"
+                        "    Global DoFs                        : %" PetscInt_FMT "\n"
+                        "    DoFs per node                      : %" PetscInt_FMT "\n"
+                        "    Global nodes                       : %" PetscInt_FMT "\n",
+                        num_P, num_Q, glob_dofs, num_comp_q, glob_dofs / num_comp_q));
+  // -- Get Partition Statistics
+  PetscCall(PetscPrintf(comm, "  Partition:                             (min,max,median,max/median)\n"));
+  {
+    PetscInt    *gather_buffer = NULL;
+    PetscInt     part_owned_dofs[3], part_local_dofs[3], part_shared_dofs[3];
+    PetscInt     median_index = comm_size % 2 ? comm_size / 2 : comm_size / 2 - 1;
+    MPI_Datatype PetscInt_MPI;
+    PetscCall(PetscDataTypeToMPIDataType(PETSC_INT, &PetscInt_MPI));
+    if (!rank) PetscCall(PetscMalloc1(comm_size, &gather_buffer));
+
+    PetscCallMPI(MPI_Gather(&owned_dofs, 1, PetscInt_MPI, gather_buffer, 1, PetscInt_MPI, 0, comm));
+    if (!rank) {
+      PetscCall(PetscSortInt(comm_size, gather_buffer));
```

Comment:
This is not expensive for GPU runs because number of ranks is not huge. For >1M ranks, it's not trivial, but I suppose still not going to be big compared to other costs.

Reply:
Yeah, I found quickselect would probably be faster, but I'd have to implement that. I figured that our problem size was small enough that it would really matter (plus it's only run once). 

Testing using numpy (used here as an upper bound), it takes about 326ms to sort 5 million 64bit ints. I'd say that's plenty fine for this usage.

Reply:
Right, you have 3 sorts and 1 second would be too much for routine small runs, but in the noise for a truly huge run like that.